### DISCUSSION TITLE
数据质量任务运行报错Permission denied

### DISCUSSION THREAD

**Discussion by yangge1017** [apache/dolphinscheduler#13531] [General]
[INFO] 2023-02-07 10:17:42.353 +0000 - Begin to pulling task
[INFO] 2023-02-07 10:17:42.354 +0000 - Begin to initialize task
[INFO] 2023-02-07 10:17:42.354 +0000 - Set task startTime: Tue Feb 07 10:17:42 UTC 2023
[INFO] 2023-02-07 10:17:42.355 +0000 - Set task envFile: /digov/soft/dolphinscheduler/worker-server/conf/dolphinscheduler_env.sh
[INFO] 2023-02-07 10:17:42.355 +0000 - Set task appId: 78_93
[INFO] 2023-02-07 10:17:42.355 +0000 - End initialize task
[INFO] 2023-02-07 10:17:42.356 +0000 - Set task status to TaskExecutionStatus{code=1, desc='running'}
[INFO] 2023-02-07 10:17:42.356 +0000 - TenantCode:root check success
[INFO] 2023-02-07 10:17:42.356 +0000 - ProcessExecDir:/tmp/dolphinscheduler/exec/process/root/7904359617056/8409814222496_14/78/93 check success
[INFO] 2023-02-07 10:17:42.356 +0000 - Resources:{} check success
[INFO] 2023-02-07 10:17:42.357 +0000 - Task plugin: DATA_QUALITY create success
[INFO] 2023-02-07 10:17:42.358 +0000 - Success initialized task plugin instance success
[INFO] 2023-02-07 10:17:42.358 +0000 - Success set taskVarPool: null
[INFO] 2023-02-07 10:17:42.359 +0000 - data quality task command: ${SPARK_HOME2}/bin/spark-submit --master local --driver-cores 1 --driver-memory 512M --num-executors 2 --executor-cores 2 --executor-memory 1G --class org.apache.dolphinscheduler.data.quality.DataQualityApplication 
[INFO] 2023-02-07 10:17:42.359 +0000 - Success create command file, command: #!/bin/bash
BASEDIR=$(cd `dirname $0`; pwd)
cd $BASEDIR
source /digov/soft/dolphinscheduler/worker-server/conf/dolphinscheduler_env.sh
${SPARK_HOME2}/bin/spark-submit --master local --driver-cores 1 --driver-memory 512M --num-executors 2 --executor-cores 2 --executor-memory 1G --class org.apache.dolphinscheduler.data.quality.DataQualityApplication 
[INFO] 2023-02-07 10:17:42.361 +0000 - task run command: sudo -u root -E bash /tmp/dolphinscheduler/exec/process/root/7904359617056/8409814222496_14/78/93/78_93.command
[INFO] 2023-02-07 10:17:42.361 +0000 - process start, process id is: 258631
[INFO] 2023-02-07 10:17:42.865 +0000 - process has exited, execute path:/tmp/dolphinscheduler/exec/process/root/7904359617056/8409814222496_14/78/93, processId:258631 ,exitStatusCode:126 ,processWaitForStatus:true ,processExitValue:126
[INFO] 2023-02-07 10:17:42.866 +0000 - Send task execute result to master, the current task status: TaskExecutionStatus{code=6, desc='failure'}
[INFO] 2023-02-07 10:17:42.867 +0000 - Remove the current task execute context from worker cache
[INFO] 2023-02-07 10:17:42.867 +0000 - The current execute mode isn't develop mode, will clear the task execute file: /tmp/dolphinscheduler/exec/process/root/7904359617056/8409814222496_14/78/93
[INFO] 2023-02-07 10:17:42.867 +0000 - Success clear the task execute file: /tmp/dolphinscheduler/exec/process/root/7904359617056/8409814222496_14/78/93
[INFO] 2023-02-07 10:17:43.362 +0000 -  -> Error: Missing application resource.
	
	Usage: spark-submit [options] <app jar | python file | R file> [app arguments]
	Usage: spark-submit --kill [submission ID] --master [spark://...]
	Usage: spark-submit --status [submission ID] --master [spark://...]
	Usage: spark-submit run-example [options] example-class [example args]
	
	Options:
	  --master MASTER_URL         spark://host:port, mesos://host:port, yarn,
	                              k8s://https://host:port, or local (Default: local[*]).
	  --deploy-mode DEPLOY_MODE   Whether to launch the driver program locally ("client") or
	                              on one of the worker machines inside the cluster ("cluster")
	                              (Default: client).
	  --class CLASS_NAME          Your application's main class (for Java / Scala apps).
	  --name NAME                 A name of your application.
	  --jars JARS                 Comma-separated list of jars to include on the driver
	                              and executor classpaths.
	  --packages                  Comma-separated list of maven coordinates of jars to include
	                              on the driver and executor classpaths. Will search the local
	                              maven repo, then maven central and any additional remote
	                              repositories given by --repositories. The format for the
	                              coordinates should be groupId:artifactId:version.
	  --exclude-packages          Comma-separated list of groupId:artifactId, to exclude while
	                              resolving the dependencies provided in --packages to avoid
	                              dependency conflicts.
	  --repositories              Comma-separated list of additional remote repositories to
	                              search for the maven coordinates given with --packages.
	  --py-files PY_FILES         Comma-separated list of .zip, .egg, or .py files to place
	                              on the PYTHONPATH for Python apps.
	  --files FILES               Comma-separated list of files to be placed in the working
	                              directory of each executor. File paths of these files
	                              in executors can be accessed via SparkFiles.get(fileName).
	
	  --conf PROP=VALUE           Arbitrary Spark configuration property.
	  --properties-file FILE      Path to a file from which to load extra properties. If not
	                              specified, this will look for conf/spark-defaults.conf.
	
	  --driver-memory MEM         Memory for driver (e.g. 1000M, 2G) (Default: 1024M).
	  --driver-java-options       Extra Java options to pass to the driver.
	  --driver-library-path       Extra library path entries to pass to the driver.
	  --driver-class-path         Extra class path entries to pass to the driver. Note that
	                              jars added with --jars are automatically included in the
	                              classpath.
	
	  --executor-memory MEM       Memory per executor (e.g. 1000M, 2G) (Default: 1G).
	
	  --proxy-user NAME           User to impersonate when submitting the application.
	                              This argument does not work with --principal / --keytab.
	
	  --help, -h                  Show this help message and exit.
	  --verbose, -v               Print additional debug output.
	  --version,                  Print the version of current Spark.
	
	 Cluster deploy mode only:
	  --driver-cores NUM          Number of cores used by the driver, only in cluster mode
	                              (Default: 1).
	
	 Spark standalone or Mesos with cluster deploy mode only:
	  --supervise                 If given, restarts the driver on failure.
	  --kill SUBMISSION_ID        If given, kills the driver specified.
	  --status SUBMISSION_ID      If given, requests the status of the driver specified.
	
	 Spark standalone and Mesos only:
	  --total-executor-cores NUM  Total cores for all executors.
	
	 Spark standalone and YARN only:
	  --executor-cores NUM        Number of cores per executor. (Default: 1 in YARN mode,
	                              or all available cores on the worker in standalone mode)
	
	 YARN-only:
	  --queue QUEUE_NAME          The YARN queue to submit to (Default: "default").
	  --num-executors NUM         Number of executors to launch (Default: 2).
	                              If dynamic allocation is enabled, the initial number of
	                              executors will be at least NUM.
	  --archives ARCHIVES         Comma separated list of archives to be extracted into the
	                              working directory of each executor.
	  --principal PRINCIPAL       Principal to be used to login to KDC, while running on
	                              secure HDFS.
	  --keytab KEYTAB             The full path to the file that contains the keytab for the
	                              principal specified above. This keytab will be copied to
	                              the node running the Application Master via the Secure
	                              Distributed Cache, for renewing the login tickets and the
	                              delegation tokens periodically.
	      
	/tmp/dolphinscheduler/exec/process/root/7904359617056/8409814222496_14/78/93/78_93.command: line 6: /digov/soft/dolphinscheduler/worker-server/libs/dolphinscheduler-data-quality-3.1.1.jar: Permission denied
[INFO] 2023-02-07 10:17:43.363 +0000 - FINALIZE_SESSION

**Comment by wsczm** [comment]
I have the same problem. Has this problem been solved

**Reply by sliencing** [reply]
把python_home配置下就好

set python_home is ok
